pkg/controllers/dynakube/version/reconciler.go

```diff
@@ -34,12 +34,33 @@ func NewReconciler(dynakube *dynatracev1beta1.DynaKube, apiReader client.Reader,
 	}
 }
 
-// Reconcile updates the version status used by the dynakube
-func (reconciler *Reconciler) Reconcile(ctx context.Context) error {
+func (reconciler *Reconciler) ReconcileCM(ctx context.Context) error {
 	updaters := []StatusUpdater{
-		newActiveGateUpdater(reconciler.dynakube, reconciler.apiReader, reconciler.dtClient, reconciler.registryClient),
-		newOneAgentUpdater(reconciler.dynakube, reconciler.apiReader, reconciler.dtClient, reconciler.registryClient),
 		newCodeModulesUpdater(reconciler.dynakube, reconciler.dtClient),
+	}
+
+	neededUpdaters := reconciler.needsReconcile(updaters)
+	if len(neededUpdaters) > 0 {
+		return reconciler.updateVersionStatuses(ctx, neededUpdaters)
+	}
+	return nil
+}
+
+func (reconciler *Reconciler) ReconcileOA(ctx context.Context) error {
+	updaters := []StatusUpdater{
+		newOneAgentUpdater(reconciler.dynakube, reconciler.apiReader, reconciler.dtClient, reconciler.registryClient),
+	}
+
+	neededUpdaters := reconciler.needsReconcile(updaters)
```

Comment:
The wording is confusing to me :) from the right side `needs` in the left `needed`?

Reply:
this is not "new new", I just moved things around here, but I agreed 😅

Reply:
trying to come up with a different name, but I am struggling 😅 

`needsReconcile(updaters)`  gives back those `updaters` that can/need-to be run.
- this is not really the nice way to do it considering I split up each updater (except for 1 instance) into its separate `Reconcile` function, but I want to minimise the refactoring, so we can tackle that later.
    - so we could probably just call [needsUpdate](https://github.com/Dynatrace/dynatrace-operator/blob/2c8bb29f220888601e0a82e1c4da5dc14a8fc8c3/pkg/controllers/dynakube/version/reconciler.go#L105-L105) on the 1 updater per `Reconcile` function

Reply:
done